enterprise/internal/campaigns/syncer.go

```diff
@@ -35,39 +39,66 @@ func (s *ChangesetSyncer) Run() {
 	if scheduleInterval == 0 {
 		scheduleInterval = 2 * time.Minute
 	}
-
-	s.queue = newChangesetQueue(100)
+	s.priorityNotify = make(chan struct{})
```

Comment:
I think it has to be buffer, otherwise the `select` almost always goes to `default` branch.
```suggestion
	s.priorityNotify = make(chan struct{}, 1)
```